src/Compiler/Driver/GraphChecking/DependencyResolution.fs

```diff
@@ -136,16 +120,16 @@ let filesInTrie (node: TrieNode) : Set<FileIndex> =
 /// - the namespace does not contain any children that can be referenced implicitly (eg. by type inference),
 /// then the main resolution algorithm does not create a link to any file defining the namespace.</para>
 /// <para>However, to satisfy the type-checker, the namespace must be resolved.
-/// This function returns a list of extra dependencies that makes sure that any such namespaces can be resolved (if it exists).
-/// For each unused open namespace we return one or more file links that define it.</para>
+/// This function returns an array with a potential extra dependencies that makes sure that any such namespaces can be resolved (if it exists).
+/// For each unused open namespace we might return one link that defined it.</para>
 /// </remarks>
 let collectGhostDependencies (fileIndex: FileIndex) (trie: TrieNode) (queryTrie: QueryTrie) (result: FileContentQueryState) =
-    // Go over all open namespaces, and assert all those links eventually went anywhere
+    // Go over all open namespaces, and assert all those links eventually went anywhere.
```

Comment:
Would help understanding the code if we clarified this comment.
What sort of assertion are we performing, if any?
It's not clear what `All those links` refers to.

IMO Something like this might be more helpful:
```
For each opened namespace, if none of already resolved dependencies define it, return the top-most file that defines it.
```